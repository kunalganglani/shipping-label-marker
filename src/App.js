import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import Home from './features/authentication/home';
import Header from './core/header';
import Login from './features/authentication/login';
import SignUp from './features/authentication/signup';
import { AuthProvider } from './features/authentication/Auth';
import PrivateRoute from './features/authentication/privateroute';

const App = () => (
  <>
    <Header headingTitle="Shipping Label Marker App" />
    <AuthProvider>
      <Router>
        <div>
          <PrivateRoute exact path="/" component={Home} />
          <Route exact path="/login" component={Login} />
          <Route exact path="/signup" component={SignUp} />
        </div>
      </Router>
    </AuthProvider>
  </>

);

export default App;
