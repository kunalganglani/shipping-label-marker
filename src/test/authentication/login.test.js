import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import App from '../../App';
test('Login renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter initialEntries={['/login']}>
        <App />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('Signup renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter initialEntries={['/signup']}>
        <App />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
