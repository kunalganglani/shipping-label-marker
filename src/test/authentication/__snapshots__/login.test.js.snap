// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
      >
        Shipping Label Marker App
      </h6>
    </div>
  </header>,
  "Loading...",
]
`;

exports[`Signup renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
      >
        Shipping Label Marker App
      </h6>
    </div>
  </header>,
  "Loading...",
]
`;
